[doc]Optimize the installation steps of the pulsar library of the Go client (#14555)
apache/pulsar: site2/docs/client-libraries-go.md

### Install go package

> You can install the `pulsar` library locally using `go get` download to local or downloading online.  

Reviewer: ```suggestion
You can get the `pulsar` library by using `go get` or use it with `go module`.  
```

Download to local:

```bash
$ go get -u "github.com/apache/pulsar-client-go/pulsar"
```

> If you can't download, please try it by configure [CDN accelerated proxy]( https://learnku.com/go/wikis/38122).

Reviewer: this should only applied to China users due the network issue, so i dont think it is proper to put it here.

Once installed locally, you can import it into your project:

```go
import "github.com/apache/pulsar-client-go/pulsar"
```

> Online download(for example):

Reviewer: ```suggestion
Use with go module:
```

### Install go package

You can get the `pulsar` library by using `go get` or use it with `go module`.  

> Download to local:

Reviewer 2: ```suggestion
Download Go client to your local environment:
```

> Download to local:

Author: Download library of Go client to your local environment:

import "github.com/apache/pulsar-client-go/pulsar"
```

Use with go module:

```bash
$ mkdir test_dir && cd test_dir 
```

> Write a sample script in 'test_dir' directory , such as ‘test_example.go’,and write ‘package main’ at the beginning of the file.

Reviewer 2: ```suggestion
Write a sample script in the `test_dir` directory (such as `test_example.go`) and write `package main` at the beginning of the file.
```
